### DISCUSSION TITLE
Auto increment column bugs

### DISCUSSION THREAD

**Discussion by kangugau** [mikro-orm/mikro-orm#4577] [General]
**Describe the bug**
When create an entity with a auto increment column, if you call flush the second time, mikro orm will make an additional update query on that column.

**To Reproduce**
1. Run this script
```ts
@Entity()
export class Customer {
  @PrimaryKey({
    type: UuidType,
  })
  uuid: string = v4();

  @Property()
  name!: string;

  @Property({ primary: false, autoincrement: true, default: 0 })
  number?: number;
}
```

```ts
  em.create(Customer, {
    name: `customer`,
  });
  await em.flush();
  // doing some stuffs...
  await em.flush();
```
2. Debug log:
 ```
[query] begin
[query] insert into `customer` (`uuid`, `name`) values ('c1e6246b-b211-4a9b-bca3-a99bbdc6c0f6', 'customer') [took 1 ms]
[query] select `c0`.`uuid`, `c0`.`number` from `customer` as `c0` where `c0`.`uuid` in ('c1e6246b-b211-4a9b-bca3-a99bbdc6c0f6') [took 2 ms]
[query] commit
[query] begin
[query] update `customer` set `number` = 2 where `uuid` = 'c1e6246b-b211-4a9b-bca3-a99bbdc6c0f6' [took 1 ms]
[query] commit
```

**Expected behavior**
Only 1 insert query will be made

**Versions**

| Dependency | Version |
| - | - |
| node | 19.7.0 |
| typescript | 4.7.4 |
| mikro-orm | 5.7.7 |
| @mikro-orm/my-sql | 5.7.7 |

**Comment by B4nan** [comment]
This should help:

https://github.com/mikro-orm/mikro-orm/commit/5d6ebe326a3e26343c268f55806a9aa97d66f439

New dev version will be out in few minutes.

**Comment by kangugau** [comment]
Tested on v5.7.15-dev.31. Thank you.
